snapcraft/internal/project_loader.py

```diff
@@ -370,20 +370,23 @@ def _build_env_for_stage(stagedir, snap_name, confinement,
     return env
 
 
-def _get_snapcraft_yaml():
-    visible_yaml_exists = os.path.exists('snapcraft.yaml')
-    hidden_yaml_exists = os.path.exists('.snapcraft.yaml')
+def get_snapcraft_yaml():
+    possible_yamls = [
+        os.path.join('snap', 'snapcraft.yaml'),
+        'snapcraft.yaml',
+        '.snapcraft.yaml',
+    ]
 
-    if visible_yaml_exists and hidden_yaml_exists:
+    snapcraft_yamls = [y for y in possible_yamls if os.path.exists(y)]
+
+    if not snapcraft_yamls:
+        raise SnapcraftYamlFileError('snap/snapcraft.yaml')
+    elif len(snapcraft_yamls) > 1:
         raise EnvironmentError(
-            "Found a 'snapcraft.yaml' and a '.snapcraft.yaml', "
-            "please remove one")
-    elif visible_yaml_exists:
-        return 'snapcraft.yaml'
-    elif hidden_yaml_exists:
-        return '.snapcraft.yaml'
-    else:
-        raise SnapcraftYamlFileError('snapcraft.yaml')
+            'Found a {!r} and a {!r}, please remove one.'.format(
+                snapcraft_yamls[0], snapcraft_yamls[1]))
```

Comment:
Good idea! But what if there is all three? They'll figure it out next time it's run?